### DISCUSSION TITLE
Layer specific infill/ top/ bottom orientation

### DISCUSSION THREAD

**Discussion by AydenHanlon** [supermerill/SuperSlicer#2035] [General]
This is probably a down the road thing but it would be neat to be able to specify infill layer direction layer by layer. So basically you would look at the gcode preview find the layer you want to change then have two small text boxes one where you type the layer you want and the other where you specify an orientation angle. used to change layer direction in Cura sometimes but being able to do it layer specific would be a game changer also being able to change the infill style. being able to set concentric infill for overhangs could help with drooping.

Wish I knew c# as I would love to be able to do this myself, sadly only know html.

**Comment by supermerill** [comment]
add a height range modifier
![image](https://user-images.githubusercontent.com/6536403/145622535-1c2c34ce-9030-4662-b944-0bc5f5250755.png)

set the start & end
![image](https://user-images.githubusercontent.com/6536403/145622639-f971423c-d984-4a8a-a065-06a016b453a1.png)

add infill->fill angle setting into it
![image](https://user-images.githubusercontent.com/6536403/145622745-63c75ccb-95d5-4a28-b041-8033fa4b8da1.png)

set your angle.
![image](https://user-images.githubusercontent.com/6536403/145622833-b51b7ad9-adfa-4e91-8afb-b40c434405df.png)


You can also change your pattern in it, and many things

Note that there may be some problem currently with the modifier for bottom detection, I'll work on it.
